api/include/opentelemetry/trace/trace_state.h

```diff
@@ -185,40 +103,37 @@ class TraceState
   std::string ToHeader()
   {
     std::string header_s;
-    for (size_t count = 0; count < num_entries_; count++)
-    {
-      if (count != 0)
-      {
-        header_s.append(",");
-      }
-
-      auto entry = (entries_.get())[count];
-      header_s.append(std::string(entry.GetKey()));
-      header_s.append(1, kKeyValueSeparator);
-      header_s.append(std::string(entry.GetValue()));
-    }
+    bool first = true;
+    kv_properties_->GetAllEntries(
+        [&header_s, &first](nostd::string_view key, nostd::string_view value) noexcept {
+          if (!first)
+          {
+            header_s.append(",");
+          }
+          else
+          {
+            first = false;
+          }
+          header_s.append(std::string(key));
```

Comment:
This is probably my mistake - but I don't think this conversion (nostd::string_view to std::string) will work properly always (eg if `key` contains end of line character). As C++11 std::string constructor won't accept std::string_view type. Better to use proper conversion : https://stackoverflow.com/questions/59424390/how-to-correctly-create-stdstring-from-a-stdstring-view